packages/messaging-api-slack/src/SlackOAuthClient.ts

```diff
@@ -354,6 +411,172 @@ export default class SlackOAuthClient {
+   */
+  _updateMessage(
+    channel: string,
+    inputMessage:
+      | {
+          text?: string;
+          attachments?: Attachment[] | string;
+          blocks?: Block[] | string;
+        }
+      | string,
+    ts: string,
+    options: UpdateMessageOptions = {}
+  ): Promise<OAuthAPIResponse> {
+    const message =
+      typeof inputMessage === 'string' ? { text: inputMessage } : inputMessage;
+
+    if (message.attachments && typeof message.attachments !== 'string') {
+      // eslint-disable-next-line no-param-reassign
+      message.attachments = JSON.stringify(
+        snakecaseKeysDeep(message.attachments)
+      );
+    }
+
+    if (message.blocks && typeof message.blocks !== 'string') {
+      // eslint-disable-next-line no-param-reassign
+      message.blocks = JSON.stringify(snakecaseKeysDeep(message.blocks));
+    }
+
+    return this.callMethod('chat.update', {
+      channel,
+      ts,
+      ...message,
+      ...options,
+    });
+  }
+
+  /**
+   * Deletes a message.
+   *
+   * https://api.slack.com/methods/chat.delete
+   */
+  _deleteMessage(
+    channel: string,
+    ts: string,
+    options: DeleteMessageOptions = {}
+  ): Promise<OAuthAPIResponse> {
+    return this.callMethod('chat.delete', {
+      channel,
+      ts,
+      ...options,
+    });
+  }
+
+  /**
+   * Share a me message into a channel.
+   *
+   * https://api.slack.com/methods/chat.meMessage
+   */
+  _meMessage(channel: string, text: string): Promise<OAuthAPIResponse> {
+    return this.callMethod('chat.meMessage', {
+      channel,
+      text,
+    });
+  }
+
+  /**
+   * Retrieve a permalink URL for a specific extant message
+   *
+   * https://api.slack.com/methods/chat.getPermalink
+   */
+  _getPermalink(channel: string, messageTs: string): Promise<OAuthAPIResponse> {
+    return this.callMethod('chat.getPermalink', {
+      channel,
+      message_ts: messageTs,
+    });
+  }
+
+  /**
+   * Schedules a message to be sent to a channel.
+   *
+   * https://api.slack.com/methods/chat.scheduleMessage
+   */
+  _scheduleMessage(
+    channel: string,
+    inputMessage:
+      | {
+          text?: string;
+          attachments?: Attachment[] | string;
+          blocks?: Block[] | string;
+        }
+      | string,
+    postAt: string,
+    options: ScheduleMessageOptions = {}
+  ): Promise<OAuthAPIResponse> {
+    const message =
+      typeof inputMessage === 'string' ? { text: inputMessage } : inputMessage;
+
+    if (message.attachments && typeof message.attachments !== 'string') {
+      // eslint-disable-next-line no-param-reassign
+      message.attachments = JSON.stringify(
+        snakecaseKeysDeep(message.attachments)
+      );
+    }
+
+    if (message.blocks && typeof message.blocks !== 'string') {
+      // eslint-disable-next-line no-param-reassign
+      message.blocks = JSON.stringify(snakecaseKeysDeep(message.blocks));
+    }
+
+    return this.callMethod('chat.scheduleMessage', {
+      channel,
+      post_at: postAt,
```

Comment:
`callMethod` will transform case for you, so you don't need to pass snakecase here.